src/test/java/com/codejoust/main/socket/RoomSocketTests.java

```diff
@@ -62,6 +63,14 @@
 
     @BeforeEach
     public void setup() throws Exception {
+        // Include the test authorization token in each request
+        template.getRestTemplate().setInterceptors(
+                Collections.singletonList((request, body, execution) -> {
+                    request.getHeaders()
+                            .add("Authorization", TestFields.TOKEN);
+                    return execution.execute(request, body);
+                }));
```

Comment:
Can you explain what's going on here - was this code partially copied from elsewhere? Is this needed for all socket tests since `template` is used rather than the normal REST testing method?

Reply:
Yeah it was from a StackOverflow post for how to add the authorization headers to TestRestTemplate (since it's different from MockMvc) - I think it's only necessary for RoomSocketTests and GameSocketTests though, whereas RoomHostTests doesn't require any auth so there's none of this code in that one